src/main/java/com/philips/research/bombase/core/scanner/domain/ScannerInteractor.java

```diff
@@ -5,25 +5,49 @@
 
 package com.philips.research.bombase.core.scanner.domain;
 
+import com.philips.research.bombase.core.downloader.DownloadService;
 import com.philips.research.bombase.core.scanner.ScannerService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 
+import java.net.URI;
 import java.nio.file.Path;
+import java.util.List;
+import java.util.stream.Collectors;
 
 @Service
 public class ScannerInteractor implements ScannerService {
+    private static final Logger LOG = LoggerFactory.getLogger(ScannerInteractor.class);
+
+    private final DownloadService downloader;
     private final ScanCodeScanner scanner;
 
-    public ScannerInteractor() {
-        this(new ScanCodeScanner());
+    @Autowired
+    public ScannerInteractor(DownloadService downloader) {
+        this(downloader, new ScanCodeScanner());
     }
 
-    ScannerInteractor(ScanCodeScanner scanner) {
+    ScannerInteractor(DownloadService downloader, ScanCodeScanner scanner) {
+        this.downloader = downloader;
         this.scanner = scanner;
     }
 
     @Override
     public ScanResult scan(Path directory) {
         return scanner.scan(directory);
     }
+
+    @Override
+    public List<String> scanLicenses(URI uri) {
+        final var licenses = downloader.download(uri, scanner::scan)
+                .getLicenses().stream()
+                .sorted((l, r) -> Integer.compare(r.getScore(), l.getScore()))
+                .peek(lic -> System.out.println(lic.getExpression() + " [" + lic.getScore() + "%]"))
```

Comment:
Is this line necessary? We already log on line 50 with LOG.info right?
```suggestion
```

Reply:
Oops; this is debugging code I forgot to remove; good catch!